### PR TITLE
DL-2462 Self heal logic fix

### PR DESCRIPTION
**Pull request by scott-goodwin**
# DL-2462 Self heal logic fix

**Bug fix**

Self heal logic fix

## Checklist

 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [x]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/services/EtmpRegimeService.scala [hunk 1]
```diff
@@ -114,13 +117,20 @@ class EtmpRegimeService @Inject()(etmpConnector: EtmpConnector,
     }
   }
 
+  def compareOptionalStrings(bcdValue: Option[String], erdValue: Option[String]): Boolean = {
+    if(bcdValue.isEmpty && erdValue.isEmpty) {
+      true
+    } else {
+      bcdValue.map(_.toUpperCase).contains(erdValue.map(_.toUpperCase).getOrElse(""))
```
Comment (jmountain-hmrc) [inline app/services/EtmpRegimeService.scala]: bcdValue.isEmpty && erdValue.isEmpty ||
  bcdValue.map(_.toUpperCase).contains(erdValue.map(_.toUpperCase).getOrElse("")

Comment (Ufuoma-Daniel) [reply app/services/EtmpRegimeService.scala]: I don't think this is an OR case. If we have 2 Nones, the right hand side of the or statement will pass and we don't want it to.

Comment (jmountain-hmrc) [reply app/services/EtmpRegimeService.scala]: The left hand statement short circuits the right hand if it's true (2 Nones)

Comment (Ufuoma-Daniel) [reply app/services/EtmpRegimeService.scala]: Oh wait, you're right.

**Pull request merged**